### PR TITLE
refactor handler/controller and their dependencied packages

### PR DESCRIPTION
**Pull request by hahattan**
the second part of removing globalvar refactor.

I was originally trying to refactor only controller and handler package, but there are many dependency packages which is hard to split from this PR.
 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-CommitMessages
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### CODE REVIEW

File: internal/controller/restfuncs.go [hunk 1]
```diff
@@ -34,56 +35,62 @@ type ConfigRespMap struct {
...
 
-func discoveryFunc(w http.ResponseWriter, req *http.Request) {
-	if checkServiceLocked(w, req) {
+func (c *RestController) discoveryFunc(w http.ResponseWriter, req *http.Request, dic *di.Container) {
+	if c.checkServiceLocked(w, req, container.DeviceServiceFrom(dic.Get).AdminState) {
 		return
 	}
-	if common.CurrentDeviceService.OperatingState == "DISABLED" {
+
+	ds := container.DeviceServiceFrom(dic.Get)
```
Comment (tonyespy) [inline internal/controller/restfuncs.go:57]: Shouldn't this come before `checkServiceLocked()` is called?

[the remark this answers is not in the document]
Comment (hahattan) [reply internal/controller/restfuncs.go:57]: you're right, updated !

**Comment by tonyespy** [comment]
@cloudxxx8 I just approved, but don't want to merge yet, as both you and Lenny are both listed as reviewers as well...

**Comment by cloudxxx8** [comment]
thanks, @tonyespy 
actually, @lenny-intel reviewed it before Chris broke to three small PRs, so I believe it is ready to be merged

**Pull request merged**